test/test_helper.exs

```diff
@@ -2,5 +2,13 @@ ExUnit.start(assert_receive_timeout: 2000)
 
 nodes = [:"first@127.0.0.1", :"second@127.0.0.1", :"third@127.0.0.1"]
 Firenest.Test.start_boot_server(hd(nodes))
-Firenest.Test.spawn_nodes(tl(nodes))
-Firenest.Test.start_firenest(nodes, adapter: Firenest.Topology.Erlang)
+Firenest.Test.start_firenest([hd(nodes)], adapter: Firenest.Topology.Erlang)
+
+# Start other nodes async, so we can start running tests that don't need them right away
+spawn_link(fn ->
+  Process.register(self(), :firenest_topology_setup)
```

Comment:
You have a race here because we can try to run this test before we even register the `firenest_topology_setup`. So I would do the registration in the parent process after `spawn_link`.